### PR TITLE
[CDAP-15787] Updated XMLMultiParser for new Validation API

### PR DESCRIPTION
**Pull request by dli357**
Updated XMLMultiParser.java for new validation API.

### CODE REVIEW

File: transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java [hunk 1]
```diff
@@ -183,42 +191,45 @@ public Config(String field, String encoding, String xPath, String schema) {
...
     }
 
-    public void validate(@Nullable Schema inputSchema) {
+    public void validate(@Nullable Schema inputSchema, FailureCollector collector) {
       if (inputSchema != null && !containsMacro(field)) {
         Schema.Field parseField = inputSchema.getField(field);
         if (parseField == null) {
-          throw new IllegalArgumentException(String.format("Field %s does not exist in the input schema.", field));
+          collector.addFailure(String.format("Field '%s' must exist in input schema.", field), null)
+              .withConfigProperty(FIELD);
```
Comment (albertshau) [inline transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java]: indentation

**Review by albertshau — APPROVED** [review]
lgtm after small indentation fixes

File: transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java [hunk 2]
```diff
@@ -183,42 +191,45 @@ public Config(String field, String encoding, String xPath, String schema) {
...
-      for (Schema.Field field : getSchema().getFields()) {
+      for (Schema.Field field : getSchema(collector).getFields()) {
         Schema fieldSchema = field.getSchema();
         Schema.Type type = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
         if (!type.isSimpleType()) {
-          throw new IllegalArgumentException(String.format(
-            "Unsupported schema. The schema can only contain simple types, but field %s is of type %s.",
-            field.getName(), type));
+          collector.addFailure(String.format("Unsupported schema type %s for field '%s'.", type, field.getName()),
+              "The schema may only contain simple types.").withOutputSchemaField(field.getName());
```
Comment (albertshau) [inline transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java]: indentation

File: transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java [hunk 3]
```diff
@@ -183,42 +191,45 @@ public Config(String field, String encoding, String xPath, String schema) {
...
     }
 
-    public XPathExpression getXPathExpression() {
+    public XPathExpression getXPathExpression(FailureCollector collector) {
       try {
         return X_PATH_FACTORY.newXPath().compile(xPath);
       } catch (XPathExpressionException e) {
-        throw new IllegalArgumentException("Could not compile xpath " + xPath);
+        collector.addFailure(String.format("Failed to compile xpath: %s", xPath), null)
+            .withConfigProperty(XPATH);
```
Comment (albertshau) [inline transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java]: indentation

File: transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java (hunk 1 above)
Comment (dli357) [reply transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java]: Fixed

File: transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java (hunk 3 above)
Comment (dli357) [reply transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java]: Fixed

File: transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java (hunk 2 above)
Comment (dli357) [reply transform-plugins/src/main/java/io/cdap/plugin/XMLMultiParser.java]: Fixed

**Pull request merged**